var/spack/repos/builtin/packages/arbor/package.py

```diff
@@ -79,55 +83,70 @@ class Arbor(CMakePackage, CudaPackage):
     # misc dependencies
     depends_on("fmt@7.1:", when="@0.5.3:")  # required by the modcc compiler
     depends_on("fmt@9.1:", when="@0.7.1:")
-    depends_on("googletest@1.12.1", when="@0.7.1:")
+    depends_on("fmt@10.2:", when="@0.9.1:")
+    depends_on("fmt@10.2:", when="@0.10.0:")
+    depends_on("googletest@1.12.1:", when="@0.7.1:")
     depends_on("pugixml@1.11:", when="@0.7.1:")
-    depends_on("nlohmann-json@3.11.2")
-    depends_on("random123")
+    depends_on("pugixml@1.13:", when="@0.9.1:")
+    depends_on("pugixml@1.14:", when="@0.10.0:")
+    depends_on("nlohmann-json@3.11.3:")
+    depends_on("random123@1.14.0:")
     with when("+cuda"):
         depends_on("cuda@10:")
         depends_on("cuda@11:", when="@0.7.1:")
+        depends_on("cuda@12:", when="@0.9.1:")
+        depends_on("cuda@12:", when="@0.10.0:")
 
     # mpi
     depends_on("mpi", when="+mpi")
     depends_on("py-mpi4py", when="+mpi+python", type=("build", "run"))
 
+    # hwloc
+    depends_on("hwloc@2:", when="+hwloc", type=("build", "run"))
+
     # python (bindings)
-    extends("python", when="+python")
-    depends_on("python@3.7:", when="+python", type=("build", "run"))
-    depends_on("py-numpy", when="+python", type=("build", "run"))
     with when("+python"):
+        extends("python")
+        depends_on("python@3.7:", type=("build", "run"))
+        depends_on("python@3.9:", when="@0.9.1:", type=("build", "run"))
+        depends_on("py-numpy", type=("build", "run"))
         depends_on("py-pybind11@2.6:", type="build")
         depends_on("py-pybind11@2.8.1:", when="@0.5.3:", type="build")
         depends_on("py-pybind11@2.10.1:", when="@0.7.1:", type="build")
+        depends_on("py-pybind11@2.10.1:", when="@0.7.1:", type="build")
+        depends_on("py-pybind11@2.10.1:", when="@2.11.1:", type="build")
```

Comment:
Interesting.  I don't see this listed in `pyproject.toml` BUT do see the header files included in `python/units.cpp` and `python/util.hpp`.